internal/release/render/render.go

```diff
@@ -197,6 +203,110 @@ func HydrateValues(release *v1alpha1.Release, mappings *config.ValueMapping) (ma
 	return result, nil
 }
 
+var objectValuesRegex = regexp.MustCompile(`^\s*\$(\w+)\(\s*((\.\w+)+)\s*\)\s*$`)
+
+const objectValuesSupportedPrefix = ".Environment.Spec.Values."
+
+func hydrateObjectValues(values map[string]any, envValues map[string]any) (map[string]any, error) {
+	resolvedValue, err := hydrateObjectValue(values, envValues)
+	if err != nil {
+		return nil, err
+	}
+	return resolvedValue.(map[string]any), err
+}
+
+func hydrateObjectValue(value any, envValues map[string]any) (any, error) {
+	switch val := value.(type) {
+	case string:
+		operator, resolvedValue, err := resolveOperatorAndValue(val, envValues)
+		if err != nil {
+			return nil, err
+		}
+		if operator != "" && operator != "ref" {
+			return nil, fmt.Errorf("only $ref() operator supported within object: %s", val)
+		}
+		return resolvedValue, nil
+	case map[string]any:
```

Comment:
probably need to duplicate the logic of `map[string]any` for `map[any]any` because the value is deserialized from yaml not json.